compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -276,19 +271,12 @@ pub fn create_program(
         (generated_acirs, generated_brillig, brillig_function_names, error_types),
         ssa_level_warnings,
     ) = optimize_into_acir(program, options)?;
-    if options.force_brillig_output {
```

Comment:
I do think this assert is correct and should remain.

Reply:
We could remove the option but case on whether we have a single main function that is marked unconstrained

Reply:
As otherwise we are going to error out when asserting the number of acir artifacts against the number of function signatures.

Reply:
We can also just update how we collect function sigs 
https://github.com/noir-lang/noir/blob/186fbb75c11fe575afd0d462f837331ccc307889/compiler/noirc_frontend/src/monomorphization/mod.rs#L146

`is_entry_point` is going to return true for functions marked with `#[fold]`. However, if we are force compiling to Brillig we do not need to collect this signature. The func sigs are only used for constructing acir artifacts. This feels like the best solution here.